spec/requests/api/v2/providers/courses_spec.rb

```diff
@@ -151,6 +151,101 @@
+    let!(:stubbed_manage_courses_api_request) do
+      stub_request(:post, %r{#{Settings.manage_api.base_url}/api/Publish/internal/course/})
+        .to_return(
+          status: api_status,
+          body: api_response
+        )
+    end
+    let(:course) {
+      create(:course,
+             provider: provider,
+             with_site_statuses: [:new],
+             with_enrichments: [:initial_draft])
+    }
+    let(:publish_path) do
+      "/api/v2/providers/#{provider.provider_code}" +
+        "/courses/#{course.course_code}/publish"
+    end
+
+    subject do
+      post publish_path, headers: { 'HTTP_AUTHORIZATION' => credentials }
+      response
+    end
+
+    context 'when unauthenticated' do
+      let(:payload) { { email: 'foo@bar' } }
+
+      it { should have_http_status(:unauthorized) }
+    end
+
+    context 'when user has not accepted terms' do
+      let(:user)         { create(:user, accept_terms_date_utc: nil) }
+      let(:organisation) { create(:organisation, users: [user]) }
+
+      it { should have_http_status(:forbidden) }
+    end
+
+    context 'when unauthorised' do
+      let(:unauthorised_user) { create(:user) }
+      let(:payload)           { { email: unauthorised_user.email } }
+
+      it "raises an error" do
+        expect { subject }.to raise_error Pundit::NotAuthorizedError
+      end
+    end
+
+    context 'when course and provider is not related' do
+      let(:course) { create(:course) }
+
+      it 'raises an error' do
+        expect { subject }.to raise_error ActiveRecord::RecordNotFound
+      end
+    end
+
+    context 'unpublished course with draft enrichment' do
+      let!(:course) {
+        create(:course,
+               provider: provider,
+               with_site_statuses: [:new],
+               with_enrichments: [:initial_draft],
+               age: 17.days.ago)
+      }
+      it 'publishes a course' do
+        # run the POST
+        response = subject
+
+        expect(course.site_statuses.first.status).to eq 'running'
+        expect(course.site_statuses.first.publish).to eq 'published'
+
+        expect(course.enrichments.first.status).to eq 'published'
+        expect(course.enrichments.first.updated_by_user_id).to eq user.id
+        expect(course.enrichments.first.updated_at).to be_within(1.second).of Time.now.utc
+        expect(course.enrichments.first.last_published_timestamp_utc).to be_within(1.second).of Time.now.utc
+        expect(course.changed_at).to be_within(1.second).of Time.now.utc
+        assert_requested :post, %r{#{Settings.manage_api.base_url}/api/Publish/internal/course/}
+        expect(response).to have_http_status(:success)
+      end
+    end
+
+    context 'when the api responds with result: false' do
+      let(:api_response) { '{ "result": false }' }
+
+      it { should have_http_status(:internal_server_error) }
+    end
+
+    context 'when the api sets http status to 500' do
+      let(:api_status)   { 500 }
```

Comment:
Is this used by search_and_compare? Is it easy to make that clearer here?

Reply:
it's actually the manage-api status. done. thanks